worlds/shorthike/Rules.py

```diff
@@ -11,8 +11,20 @@ def create_rules(self, location_table):
             forbid_items_for_player(multiworld.get_location(loc["name"], player), self.item_name_groups['Maps'], player)
             add_rule(multiworld.get_location(loc["name"], player),
                 lambda state: state.has("Shovel", player))
+
+        # Shop Rules
         if loc["purchase"] and not options.coins_in_shops:
             forbid_items_for_player(multiworld.get_location(loc["name"], player), self.item_name_groups['Coins'], player)
+        if loc["purchase"] >= get_min_shop_logic_cost(self) and options.shop_check_logic != 0:
+            if options.shop_check_logic == 1 or options.shop_check_logic == 2:
```

Comment:
For readability purposes, please compare either against strings `if options.shop_check_logic == "fishing_rod"` or, most ideally, against the the field itself `if options.shop_check_logic == ShopCheckLogic.option_fishing_rod`

Reply:
There are multiple instances of this, please check them all :)

Reply:
Fixed this 👍